### PR TITLE
Fixes in iron: union, intersection, Not[Empty] and other cases

### PR DESCRIPTION
**Pull request by kciesielski**
Fixes https://github.com/softwaremill/tapir/issues/3830

This PR fixes some bugs in iron integration:
* union and intersection mirrors were using dealiased types, causing predicates like `MinLength` become `DescribedAs[Length[GreaterEqual[0]]], _]`. This made our validator resolution fail.
* union and intersection primitive elements had an issue in `summonFrom`, resulting in wrong validators being resolved

These fixes should cover cases of problematic schema resolution for types with constrains like `DescribedAs`, `String :| Not[Empty]`, unions, intersections, and possibly some other cases.

### CODE REVIEW

File: integrations/iron/src/main/scala/sttp/iron/codec/iron/TapirCodecIron.scala [hunk 1]
```diff
@@ -133,8 +133,8 @@ trait TapirCodecIron extends DescriptionWitness with LowPriorityValidatorForPred
       case _: EmptyTuple => Nil
       case _: (head *: tail) =>
         val headValidator: ValidatorForPredicate[N, ?] = summonFrom {
-          case pv: PrimitiveValidatorForPredicate[N, head] => pv
-          case _                                           => summonInline[ValidatorForPredicate[N, head]]
+          case pv: PrimitiveValidatorForPredicate[N, `head`] => pv
```
Comment (kciesielski) [inline integrations/iron/src/main/scala/sttp/iron/codec/iron/TapirCodecIron.scala:136]: This one was the nastiest :angry:

File: integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala [hunk 2]
```diff
@@ -33,7 +33,7 @@ object IntersectionTypeMirror {
     def rec(tpe: TypeRepr): TypeRepr = {
       tpe.dealias match
         case AndType(left, right) => concatTypes(rec(left), rec(right))
-        case t                    => prependTypes(t, TypeRepr.of[EmptyTuple])
+        case t                    => prependTypes(tpe, TypeRepr.of[EmptyTuple])
```
Comment (adamw) [inline integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala:36]: do you know *why* you have to pass the aliased type here? if so, maybe it's worth a comment?

**Review by adamw — APPROVED** [review]

File: integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala (hunk 2 above)
Comment (kciesielski) [reply integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala:36]: Yes, I mentioned it in the PR description, but I can add a  more detailed explanation here as a comment.

Comment (adamw) [reply integrations/iron/src/main/scala/sttp/iron/codec/iron/IntersectionTypeMirror.scala:36]: Ah, I forgot to read the description. Yeah, in the code would be much better, for future generations :)

**Pull request merged**